TrafficCapture/coreUtilities/src/main/java/org/opensearch/migrations/tracing/RootOtelContext.java

```diff
@@ -0,0 +1,139 @@
+package org.opensearch.migrations.tracing;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.MeterProvider;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.exporter.otlp.logs.OtlpGrpcLogRecordExporter;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporter;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.logs.SdkLoggerProvider;
+import io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.opensearch.migrations.Utils;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+public class RootOtelContext implements IRootOtelContext {
+    private final OpenTelemetry openTelemetryImpl;
+    private final String scopeName;
+    @Getter
+    @Setter
+    Exception observedExceptionToIncludeInMetrics;
+
+    public static OpenTelemetry initializeOpenTelemetryForCollector(@NonNull String collectorEndpoint,
+                                                                    @NonNull String serviceName) {
+        var serviceResource = Resource.getDefault().toBuilder()
+                .put(ResourceAttributes.SERVICE_NAME, serviceName)
+                .build();
+
+        final var spanProcessor = BatchSpanProcessor.builder(OtlpGrpcSpanExporter.builder()
+                        .setEndpoint(collectorEndpoint)
+                        .setTimeout(2, TimeUnit.SECONDS)
+                        .build())
+                .setScheduleDelay(100, TimeUnit.MILLISECONDS)
```

Comment:
Is there a reason we may want the span's processed at a 10x faster interval than the metrics?

Reply:
Good point.
I'm not sure why I clocked that so tightly.  The default is 5s.  I'll kick it back to that in case there's a way eventually load that via an environment variable, etc.